### PR TITLE
feat(decimal): display full 8-decimal amounts with BigInt precision

### PR DESCRIPTION
**Pull request by raul-oliveira**
### Acceptance Criteria
The decimal precision changed from 2 to 18 places. The explorer must display these amounts fully and correctly. This PR addresses the frontend so the following screens render full-precision values:

- **Transaction detail**: input, output and fee-paid amounts (overview).
- **Address detail**: total received, total spent, unlocked/locked balance, and the transaction listing (already right-aligned so growing values stay aligned).
- **Token balances**: total, unlocked and locked per address.
- **Token detail**: total supply.

**Root cause & fix.** The display layer was already decimal-aware (`prettyValue` reads `decimal_places` from the node's version API and already supports `BigInt`). The real defect was precision loss upstream: in full mode, Explorer Service responses were parsed with the default `JSON.parse`, which silently rounds any integer above `Number.MAX_SAFE_INTEGER` (~0.009 token at 18 decimals). This PR:

- Parses Explorer Service responses with wallet-lib's `bigIntUtils.JSONBigInt` at the axios boundary, so out-of-safe-range integers stay exact (`BigInt`) while timestamps/heights/counts remain `Number`. Falls back gracefully for empty/non-JSON bodies (e.g. graphviz).
- Makes value arithmetic `BigInt`-safe: address "Total spent" (`total_received - unlocked - locked`), legacy "Final balance" (`received - spent`), and the legacy per-tx balance accumulation (`calculateAddressBalance`).
- Fixes a regression the broader `BigInt` parsing introduces: the nano-contract logs viewer used native `JSON.stringify` (which throws on `BigInt`), now switched to `JSONBigInt.stringify`.
- Adds the repo's first unit tests (parsing, arithmetic, balance calc) and a jest `env` override for test files.

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. No new dependencies — reuses wallet-lib's existing `bigIntUtils`.

### Screenshots
image
image
image
image
image
image
image
image

## Summary by CodeRabbit

- **Bug Fixes**
  - Preserved full integer precision when parsing API responses and transaction data, keeping large balances and values accurate.
  - Improved address balance and spending calculations by centralizing BigInt-safe math and correctly excluding authority outputs.
  - Updated Nano contract log JSON rendering to use BigInt-aware stringify for more accurate display.
- **Tests**
  - Added and expanded Jest coverage for BigInt JSON parsing, balance computation, and BigInt-safe subtraction (including mixed numeric inputs and boundary values).

### CODE REVIEW

File: src/screens/nano/NanoContractLogs.js [hunk 1]
```diff
@@ -80,7 +81,7 @@ function NanoContractLogs() {
         <div className="blueprint-source-code">
           <pre>
             <code ref={logsRef} className="language-json">
-              {JSON.stringify(logsResponse.logs, null, 2)}
+              {hathorLib.bigIntUtils.JSONBigInt.stringify(logsResponse.logs, 2)}
```
[the remark this answers is not in the document]
Comment (raul-oliveira) [reply src/screens/nano/NanoContractLogs.js:84]: Not applying — this would break the indentation.

`hathorLib.bigIntUtils.JSONBigInt.stringify` has the signature `stringify(value, space)` and manages the replacer internally (`JSON.stringify(value, this.bigIntReplacer, space)`), unlike the native `JSON.stringify(value, replacer, space)`. So `stringify(logs, 2)` already passes `2` as `space` and keeps the multi-line layout. The proposed `stringify(logs, null, 2)` would set `space = null` (the `2` becomes an ignored 3rd argument), collapsing the JSON to a single line.

Verified at runtime:
- `JSONBigInt.stringify({a:1,b:{c:2n}}, 2)` → multi-line (contains `\n`)
- `JSONBigInt.stringify({a:1,b:{c:2n}}, null, 2)` → single line (no `\n`)

Keeping the current code.

File: src/utils/addressBalance.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    if (txin.token === token && txin.decoded.address === address && !isAuthorityOutput(txin)) {
+      value -= BigInt(txin.value);
+    }
+  }
+
+  for (const txout of tx.outputs) {
+    if (txout.token === token && txout.decoded.address === address && !isAuthorityOutput(txout)) {
+      value += BigInt(txout.value);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (raul-oliveira) [reply src/utils/addressBalance.js:35]: Fixed in fa4459b — `decoded?.address` in both the input and output loops, consistent with the existing guard in `AddressDetailExplorer.js:41`. Added a test case covering inputs/outputs without a decoded script.

File: src/utils/valueMath.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * loses precision above 2^53 for Number. This normalizes every operand to BigInt
+ * so the result is always exact.
+ *
+ * @param {number|bigint|string} minuend Value to subtract from
+ * @param {...(number|bigint|string)} subtrahends Values to subtract
+ * @return {bigint} The exact difference
+ */
+export function subtractValues(minuend, ...subtrahends) {
+  return subtrahends.reduce((acc, value) => acc - BigInt(value), BigInt(minuend));
+}
```
[the remark this answers is not in the document]
Comment (raul-oliveira) [reply src/utils/valueMath.js:22]: Not applying. At both call sites the operands are balance fields (`total_received`, `unlocked_balance`, `locked_balance`, and legacy `received`/`spent`), and those exact fields are also rendered directly via `renderValue → prettyValue → BigInt(...)` on the same screen (e.g. `AddressSummary` lines 93/107/111). If any were nullish, the render would already throw there — `subtractValues` is not the crash site. The explorer-service `address/balance` response always includes these as integer fields (Python `int` from the `BIGINT UNSIGNED` columns), so nullish does not occur in practice.

A silent `?? 0` here would only mask a genuine data problem while the direct `prettyValue` calls would still throw. If we want defensive handling, it should be applied consistently upstream rather than as a silent zero in the math utility. Leaving as-is.
